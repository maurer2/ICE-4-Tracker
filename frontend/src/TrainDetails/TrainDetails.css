.information {
  margin: 0;
  background: #fff;
  color: black;
  text-align: left;
}
