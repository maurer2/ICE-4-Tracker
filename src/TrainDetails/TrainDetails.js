import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './TrainDetails.module.css';

class TrainDetails extends Component {
  render() {
    const { selectedTrainNumber, trainDetails } = this.props;
    const trainName = `ICE-${selectedTrainNumber}`;

    return (
      <div className={ styles['train-details'] }>
        <h2 className={ styles.title }>
          Informationen zu <strong>{ trainName }</strong>
        </h2>
        <dl className={ styles.information }>
          <dt className={ styles.number }>Nummer:</dt>
          <dd className={ styles.value }>{ trainName }</dd>

          <span className={ styles.breaker }></span>

          <dt className={ styles.origin }>Start:</dt>
          <dd className={ styles.value }>{ trainDetails.origin }</dd>

          <span className={ styles.breaker }></span>

          <dt className={ styles.destination }>Ziel:</dt>
          <dd className={ styles.value }>{ trainDetails.destination }</dd>
        </dl>
      </div>
    );
  }
}

export default TrainDetails;

TrainDetails.propTypes = {
  selectedTrainNumber: PropTypes.string.isRequired,
  trainDetails: PropTypes.object.isRequired,
};
