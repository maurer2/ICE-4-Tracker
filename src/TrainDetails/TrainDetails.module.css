.information {
  display: flex;
  margin: 0;
  padding: 0.5rem;
  flex-wrap: wrap;
  text-align: left;
  background: #fff;
  color: black;
}

.title {
  margin: 0 0 1rem 0;
  font-size: 2rem;
  color: white;
}

.number,
.origin,
.destination {
  margin-left: 0;
}

.value {
  margin-left: 0.25rem;
}

.breaker {
  width: 100%;
}
