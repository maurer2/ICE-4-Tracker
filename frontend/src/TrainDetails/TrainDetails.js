import React, { Component } from 'react';
import './TrainDetails.css';

class TrainDetails extends Component {
  render(){
    return (
      <div className="train-details">
        <h2 className="subtitle">
          Zug-Informationen
        </h2>
        <dl className="information">
          <dt className="train-number">Nummer</dt>
          <dd className="train-number-value">ICE-555</dd>

          <dt className="train-origin">Start</dt>
          <dd className="train-origin-value">Stadt 1</dd>

          <dt className="train-destination">Ziel</dt>
          <dd className="train-destination-value">Stadt 2</dd>
        </dl>
      </div>
    )
  }
}

export default TrainDetails;
