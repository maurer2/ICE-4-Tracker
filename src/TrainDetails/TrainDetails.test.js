import React from 'react';
import TrainDetails from './TrainDetails';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('TrainDetails', () => {
  const dummyTrainDetails = {
    trainNumber: '502',
    origin: 'Mï¿½nchen Hbf',
    destination: 'Hamburg-Altona/Hmb Hbf'
  }

  const wrapper = shallow(
    <TrainDetails
      selectedTrainNumber={ '500' }
      trainDetails={ dummyTrainDetails }
    />
  );

  test('TrainDetails should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
