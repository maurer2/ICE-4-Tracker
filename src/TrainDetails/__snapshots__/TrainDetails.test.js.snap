// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainDetails TrainDetails should match snapshot 1`] = `
<div
  className="train-details"
>
  <h2
    className="title"
  >
    Informationen zu 
    <strong>
      ICE-500
    </strong>
  </h2>
  <dl
    className="information"
  >
    <dt
      className="number"
    >
      Nummer:
    </dt>
    <dd
      className="value"
    >
      ICE-500
    </dd>
    <span
      className="breaker"
    />
    <dt
      className="origin"
    >
      Start:
    </dt>
    <dd
      className="value"
    >
      Mï¿½nchen Hbf
    </dd>
    <span
      className="breaker"
    />
    <dt
      className="destination"
    >
      Ziel:
    </dt>
    <dd
      className="value"
    >
      Hamburg-Altona/Hmb Hbf
    </dd>
  </dl>
</div>
`;
